src/com/owncloud/android/files/FileOperationsHelper.java

```diff
@@ -88,13 +93,28 @@ public void openFile(OCFile file) {
             }
             
             Intent chooserIntent;
+            List<ResolveInfo> launchables;
             if (intentForGuessedMimeType != null) {
                 chooserIntent = Intent.createChooser(intentForGuessedMimeType, mFileActivity.getString(R.string.actionbar_open_with));
+                launchables = mFileActivity.getPackageManager().queryIntentActivities(intentForGuessedMimeType, PackageManager.GET_INTENT_FILTERS);
             } else {
                 chooserIntent = Intent.createChooser(intentForSavedMimeType, mFileActivity.getString(R.string.actionbar_open_with));
+                launchables = mFileActivity.getPackageManager().queryIntentActivities(intentForSavedMimeType, PackageManager.GET_INTENT_FILTERS);
             }
```

Comment:
As this is nearly 100% the same I would use a variable for intentForGuessedMimeType/intentForSavedMimeType and remove the duplicate code.